design-documents/batch-query-services.md

```diff
@@ -0,0 +1,77 @@
+#Batch query services
+
+## Purpose
+
+The main reason for such change is performance.
+The system triggers multiple queries during an operation (page load, service call). 
+Instead of making several requests for data retrieval,
+it would be more efficient to assemble all requests and send them as the single call to query service.
+Even without internal code optimization such approach saves time that required for the service components bootstrap.
+* Such an approach saves time that required for the service components bootstrap.
+* The service implementation can be changed to guarantee better performance for batch operations.
+
+## Arguments of operation
+
+A service that manipulates with a single request could have a straightforward interface:
+```php
+    /**
+     * @param int $productId
```

Comment:
should be int[] type

Reply:
Not exactly correct. actually, this proposal about how to satisfy multiple clients of a single service. 
Arguments segmentation really matters in this case.

Reply:
So, do you think that this service should not use batch? Looks like our Technical Vision say that all services should be batched